src/server/src/routes/student/index.ts

```diff
@@ -142,6 +143,68 @@ router.get(
     }
 );
 
+/******************************************************************************
+ *            GET Request - Search - /api/student/search"
+ ******************************************************************************/
+
+interface StudentSearchRequest extends Request {
+    query: {
+        firstName?: string;
+        lastName?: string;
+        email?: string;
+        sid?: string;
+        departmentIds?: string[];
+        classStandings?: classStandings[];
+        page: string;
+        numOfItems: string;
+    };
+}
+
+router.get(
+    '/search',
+    passport.authenticate('jwt', { session: false }),
+    validationMiddleware({ querySchema: studentSearchSchema }),
+    async (req: StudentSearchRequest, res: Response) => {
+        const { departmentIds, page, numOfItems } = req.query;
+        let { firstName, lastName, email, sid, classStandings } = req.query;
+
+        if (!firstName) firstName = "";
+        if (!lastName) lastName = "";
+        if (!email) email = "";
+        if (!sid) sid = "";
+        var departmentIdInts = [];
+        // Pass -1 when the input is empty because it causes a sql parse error
+        // when we pass in an empty array.
+        if (!departmentIds) departmentIdInts = [-1];
```

Comment:
The client will never send departmentIds as undefined, so I don't think we need this. Note: for the client optional for departmentIds means an empty array.

Reply:
But if the departmentIds array is empty, it causes an SQL parse error. That's why I added -1. What should we do?

Reply:
Done.